HierarchicalDataConverter/Catalyst.HierarchicalDataConverter.AutomatedTests/GrandparentParentChildTest.cs

```diff
@@ -0,0 +1,441 @@
+                                   Status = FieldStatus.Omitted
+                               },
+                           new Field
+                               {
+                                   FieldName = "Level2FieldToBeAdded",
+                                   Status = FieldStatus.Active
+                               }
+                       };
+        }
+
+        private Binding GetNestedBindingLevel0Source()
+        {
+            var destinationEntity = this.GetNestedDestinationEntity();
+            var nestedBindingLevel1 = this.GetNestedBindingLevel1Source();
+            var nestedBindingLevel2 = this.GetNestedBindingLevel2Source();
+            var level0SourceEntity = this.GetLevel0SourceEntity();
+            return new Binding
+                       {
+                           DestinationEntityId = destinationEntity.Id,
+                           ContentId = Guid.NewGuid(),
+                           DataMartId = this.DataMartId,
+                           Classification = "Generic",
+                           Name = "NestedBindingLevel0Source",
+                           SourceConnectionId = this.SourceConnectionId,
+                           BindingType = "Nested",
+                           Status = "Active",
+                           LoadTypeCode = "Full",
+                           Id = 0,
+                           SourcedByEntities = { new SourceEntityReference { SourceEntityId = level0SourceEntity.Id } },
+                           ObjectRelationships =
+                               {
+                                   new ObjectReference
+                                       {
+                                           ChildObjectId =
+                                               nestedBindingLevel1.Id,
+                                           ChildObjectType = "Binding",
+                                           AttributeValues =
+                                               {
+                                                   new
+                                                   ObjectAttributeValue
+                                                       {
+                                                           AttributeName
+                                                               = "Cardinality",
+                                                           AttributeValue
+                                                               = "SingleObject"
+                                                       },
+                                                   new
+                                                   ObjectAttributeValue
+                                                       {
+                                                           AttributeName
+                                                               = "ParentKeyFields",
+                                                           AttributeValue
```

Comment:
I don't see these columns in any of the GetLevelXXFields() methods.  I think that the all three Source entities need to have 'id0' and both level1 and level2 need to have 'id1' in order for all of these relationships to make sense.